web-services-app/src/test/java/edu/unc/lib/boxc/web/services/rest/SingleUseKeyControllerIT.java

```diff
@@ -0,0 +1,178 @@
+                .andReturn();
+    }
+
+    @Test
+    public void testGenerateFailure() throws Exception {
+        var id = pid.getUUID();
+        var fileObject = repositoryObjectFactory.createFileObject(pid, null);
+        when(repositoryObjectLoader.getRepositoryObject(eq(pid))).thenReturn(fileObject);
+        singleUseKeyService.setCsvPath(null);
+
+        mvc.perform(post("/single_use_link/create/" + id))
+                .andExpect(status().isInternalServerError())
+                .andReturn();
+    }
+
+    @Test
+    public void testGenerateSuccess() throws Exception {
+        var fileObject = repositoryObjectFactory.createFileObject(pid, null);
+        when(repositoryObjectLoader.getRepositoryObject(eq(pid))).thenReturn(fileObject);
+
+        MvcResult result = mvc.perform(post("/single_use_link/create/" + pid.getUUID()))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+
+        Map<String, Object> respMap = MvcTestHelpers.getMapFromResponse(result);
+        assertNotNull(respMap.get("url"));
+        assertEquals(pid.getUUID(), respMap.get("target_id"));
+        assertNotNull(respMap.get("expires"));
+    }
+
+    @Test
+    public void testDownloadAccessKeyInvalidCsvDoesNotExist() throws Exception {
+        var accessKey = SingleUseKeyService.getKey();
+        mvc.perform(get("/single_use_link/" + accessKey))
+                .andExpect(status().isNotFound())
+                .andReturn();
+    }
+
+    @Test
+    public void testDownloadAccessKeyIsNotInCsv() throws Exception {
+        var accessKey = SingleUseKeyService.getKey();
+        var expirationTimestamp = System.currentTimeMillis() + DAY_MILLISECONDS;
+        generateDefaultCsv(csvPath, null, expirationTimestamp);
+        mvc.perform(get("/single_use_link/" + accessKey))
+                .andExpect(status().isNotFound())
+                .andReturn();
+    }
+
+    @Test
+    public void testDownloadSuccess() throws Exception {
+        var content = "binary content";
+        var filePid = PIDs.get(UUID_TEST);
+        FileObject fileObj = repositoryObjectFactory.createFileObject(filePid, null);
+        Path contentPath = Files.createTempFile("file", ".txt");
+        FileUtils.writeStringToFile(contentPath.toFile(), content, "UTF-8");
+        fileObj.addOriginalFile(contentPath.toUri(), "file.txt", "text/plain", null, null);
+
+        var accessKey = SingleUseKeyService.getKey();
+        var expirationTimestamp = System.currentTimeMillis() + DAY_MILLISECONDS;
+        generateDefaultCsv(csvPath, accessKey, expirationTimestamp);
+        MvcResult result = mvc.perform(get("/single_use_link/" + accessKey))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+
+        MockHttpServletResponse response = result.getResponse();
+        assertEquals(content, response.getContentAsString());
+
+        assertEquals(content.length(), response.getContentLength());
```

Comment:
Since we're doing "equals" at line 172, I don't think we need to compare the response length here.